src/Sequence.cc

```diff
@@ -388,10 +388,10 @@ Sequence::execute_sequence_impl(Iterator step_begin, Iterator step_end, Context&
 {
     Iterator step = step_begin;
 
-    if (comm and comm->immediate_termination_requested_)
-        throw Error{ gul14::cat(abort_marker, "Stop on user request") };
     while (step < step_end)
     {
+        if (comm and comm->immediate_termination_requested_)
+            throw Error{ gul14::cat(abort_marker, "Stop on user request") };
```

Comment:
We should use the chance to add a step index to the exception (`step - steps_.begin()`). Even for a user abort, this may sometimes give some interesting information.

Reply:
Bah, I thought noone would notice. Now I have to think about if the formula return the correct index always ;)

Reply:
I think a Step Index may/should not be mentioned before that Step messaged 'Start', so we message the previous Step's index or nothing.
Check if that is useful.